src/AppInstallerCommonCore/Filesystem.cpp

```diff
@@ -238,4 +238,18 @@ namespace AppInstaller::Filesystem
         THROW_IF_FAILED(SHGetKnownFolderPath(id, KF_FLAG_NO_ALIAS | KF_FLAG_DONT_VERIFY | KF_FLAG_NO_PACKAGE_REDIRECTION, NULL, &knownFolder));
         return knownFolder.get();
     }
+
+    bool IsSameVolume(const std::filesystem::path& current, const std::filesystem::path& target)
+    {
+        LPWSTR currentVolumeBuffer = new WCHAR[MAX_PATH];

```

Comment:
new without delete is problematic

Use std::wstring as buffer like what we did in other places, and set buffer length to input length